extras/indexingExample/src/main/java/AccumuloRyaDetailsExample.java

```diff
@@ -0,0 +1,154 @@
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails.PCJDetails.PCJUpdateStrategy;
+import mvm.rya.api.instance.RyaDetails.ProspectorDetails;
+import mvm.rya.api.instance.RyaDetails.TemporalIndexDetails;
+import mvm.rya.api.instance.RyaDetailsToConfiguration;
+import mvm.rya.api.persist.RyaDAOException;
+import mvm.rya.indexing.accumulo.ConfigUtils;
+import mvm.rya.indexing.accumulo.geo.GeoConstants;
+import mvm.rya.sail.config.RyaSailFactory;
+
+public class AccumuloRyaDetailsExample {
+    private static final Logger log = Logger.getLogger(AccumuloRyaDetailsExample.class);
+
+    //
+    // Connection configuration parameters
+    //
+
+    private static final boolean USE_MOCK_INSTANCE = true;
+    private static final boolean PRINT_QUERIES = true;
+    private static final String INSTANCE = "instance";
+    private static final String RYA_TABLE_PREFIX = "x_test_triplestore_";
+    private static final String AUTHS = "U";
+
+    public static void main(final String[] args) throws Exception {
+        final Configuration conf = getConf();
+        conf.setBoolean(ConfigUtils.DISPLAY_QUERY_PLAN, PRINT_QUERIES);
+
+        final RyaDetails.Builder builder = RyaDetails.builder();
+
+        builder.setRyaInstanceName(INSTANCE)
+            .setRyaVersion("1.2.3.4")
+            .setEntityCentricIndexDetails( new EntityCentricIndexDetails(true) )
+            .setGeoIndexDetails( new GeoIndexDetails(false) )
+            .setTemporalIndexDetails( new TemporalIndexDetails(true) )
+            .setFreeTextDetails( new FreeTextIndexDetails(true) )
+            .setPCJIndexDetails(
+                    PCJIndexDetails.builder()
+                        .setEnabled(true)
+                        .setFluoDetails( new FluoDetails("test_instance_rya_pcj_updater") )
+                        .addPCJDetails(
+                                PCJDetails.builder()
+                                    .setId("pcj 1")
+                                    .setUpdateStrategy(PCJUpdateStrategy.BATCH)
+                                    .setLastUpdateTime( new Date() )
+                                    .build())
+                        .addPCJDetails(
+                                PCJDetails.builder()
+                                    .setId("pcj 2")
+                                    .setUpdateStrategy(PCJUpdateStrategy.INCREMENTAL)
+                                    .build())
+                        .build())
+            .setProspectorDetails( new ProspectorDetails(Optional.of(new Date())) )
+            .setJoinSelectivityDetails( new JoinSelectivityDetails(Optional.of(new Date())) );
+        RyaDetailsToConfiguration.addRyaDetailsToConfiguration(builder.build(), conf);
+
+        SailRepository repository = null;
+        SailRepositoryConnection conn = null;
+        try {
+            log.info("Connecting to Indexing Sail Repository.");
+            final Sail sail = RyaSailFactory.getInstance(conf);
```

Comment:
RyaSailFactory should use your RyaDetailsToConfiguration object to merge the user's configuration with the RyaDetails for the specified Rya prefix.  To get the RyaDetails for the instance, the RyaSailFactory should first determine whether the user wants to use Mongo or Accumulo.  Based on that configuration, the Factory should initialize the appropriate RyaDetailsRepository to fetch the details for merging the user's config with the prefix config.  Again, this should all happen within the RyaSailFactory and be hidden from the user.